docs/scripting-element-object.rst

```diff
@@ -0,0 +1,566 @@
+reflected in page source you must call :ref:`splash-wait`
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-mouse-hover:
+
+element:mouse_hover
+-------------------
+
+Trigger mouse hover (JavaScript mouseover) event on the element.
+
+**Signature:** ``ok, reason = element:mouse_hover{x=0, y=0}``
+
+**Parameters:**
+
+* x - optional, x coordinate relative to the left corner of the element
+* y - optional, y coordinate relative to the top corner of the element
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+If x or y coordinate is not provided they will be set to 0 and the click will be triggered
+on the left-top corner of the element. The coordinates can have a negative value which means
+the click will be triggered outside of the element.
+
+Mouse events are not propagated immediately, to see consequences of click
+reflected in page source you must call :ref:`splash-wait`
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-styles:
+
+element:styles
+--------------
+
+Return the computed styles of the element.
+
+**Signature:** ``styles = element:styles()``
+
+**Returns:** ``styles`` is a table with computed styles of the element.
+
+**Async:** no.
+
+Example of getting the font size of the element using this method.
+
+.. code-block:: lua
+
+    function main(splash)
+        local element = splash:select('.element')
+        return element:styles()['font-size']
+    end
+
+
+.. _splash-element-bounds:
+
+element:bounds
+--------------
+
+Return the bounding client rectangle of the element
+
+**Signature:** ``bounds = element:bounds()``
+
+**Returns:** ``bounds`` is a table with the client bounding rectangle with the ``top``, ``right``,
+``bottom`` and ``left`` coordinates.
+
+**Async:** no.
+
+Example of getting the bounds of the element.
+
+.. code-block:: lua
+
+    function main(splash)
+        local element = splash:select('.element')
+        return element:bounds()
+        -- e.g. bounds is { top = 10, right = 20, bottom = 20, left = 10 }
+    end
+
+
+.. _splash-element-png:
+
+element:png
+-----------
+
+Return a screenshot of the element in PNG format
+
+**Signature:** ``ok, shot = element:png{width=nil, scale_method='raster', pad=0}``
+
+**Parameters:**
+
+* width - optional, width of a screenshot in pixels;
+* scale_method - optional, method to use when resizing the image, ``'raster'``
+  or ``'vector'``;
+* pad - optional, integer or ``{left, top, right, bottom}`` values of padding
+
+**Returns:** ``ok, shot`` pair. If ``ok`` is nil then error happened during the
+function call; ``shot`` provides an information about error type; otherwise
+``shot`` is a PNG screenshot data, as a :ref:`binary object <binary-objects>`.
+When the result is empty (e.g. if the element is not visible) ``nil`` is returned.
```

Comment:
What do you think about removing ok flag for this function? It looks like for practical purposes there is no much difference between invisible HTML element and an element which no longer present on a page.